osp/osp.h

```diff
@@ -77,6 +77,13 @@ osp_get_vts_version (osp_connection_t *, char **);
 int
 osp_get_vts (osp_connection_t *, entity_t *);
 
+typedef struct {
+  char *filter; ///< the filter to apply for a vt sub-selection.
+} osp_get_vts_opts_t;
```

Comment:
Just for curiosity, why do are the typedefs put into the headers? I've seen that a lot in the past in our code base. Personally I never would make them public.

Reply:
I followed the scheme of osp_start_scan_ext().

The calling module needs to fill the "opts" properly. So it needs to know about the single
elements of it.
See here how I used it: 
https://github.com/greenbone/gvmd/pull/648/commits/691a6d2ed5705d554c11af632ff60aa191ddeea7

Reply:
That's exactly the issue i have with making structs public.

>get_vts_opts.filter = g_strdup_printf ("modification_time>%s", db_feed_version); 

There should be a public function like `void osp_get_vts_opts_set_filter(osp_get_vts_opts_t *opts, const char *filter)` to not let users access internals like this struct. To be clear the layout of the data in the memory should be an implementation detail and not directly accessible by users of the api.

Reply:
The scope of my patch is to aid the the patch about acceleration of the feed update.
So I applied the scheme that is done for all similar OSP and GMP API functions (about 10 of them I found) rather than introducing a second scheme for just one function.

Reply:
I was not asking about changing your patch. I was just curios about the reason why we make data structs public accessible. Personally I am not going to change that at all.

Reply:
In this case we're following the scheme introduced by someone from long ago.
Can see how your suggestion would be cleaner (but more work).